### PR TITLE
Improve & cleanup documentation comments

### PR DESCRIPTION
**Pull request by cr7pt0gr4ph7**
This PR contains multiple small improvements to wording and formatting of the documentation comments.

If desired, I can squash the commits into one, although I think that the commits are fine as-is.

### CODE REVIEW

**Comment by cr7pt0gr4ph7** [comment]
Rebased onto newest master.

**Comment by cr7pt0gr4ph7** [comment]
Rebased.

File: api/types/client.go [hunk 1]
```diff
@@ -154,28 +154,19 @@ type ImageBuildResponse struct {
 
 // ImageCreateOptions holds information to create images.
 type ImageCreateOptions struct {
-	// Parent is the image to create this image from
-	Parent string
-	// Tag is the name to tag this image
-	Tag string
-	// RegistryAuth is the base64 encoded credentials for this server
-	RegistryAuth string
+	Parent       string // Parent is the image to create this image from
```
Comment (thaJeztah) [inline api/types/client.go]: just a question; does the positioning of the comment make a difference for GoDoc?

**Comment by thaJeztah** [comment]
One question, but I like the reformatting. I do think the commits can be squashed though

**Comment by thaJeztah** [comment]
ping @calavera for review :)

**Comment by cr7pt0gr4ph7** [comment]
I'll wait with squashing until after the review.

**Comment by thaJeztah** [comment]
when rebasing/squashing, yes, but I don't see merge conflicts yet :)

File: api/types/client.go (hunk 1 above)
Comment (cr7pt0gr4ph7) [inline api/types/client.go]: AFAIK, GoDoc does not even care where the comment is positioned; the body of the struct type is just printed out as-is ([example from `net/http`](https://godoc.org/net/http#Response) that contains both positions).
The standard library uses both positions, dependending on the length of the comment.

Where positioning might matter is IDE support. As far as I can remember, though, the Go plugin for IntelliJ IDEA handles both positions correctly.

Comment (thaJeztah) [inline api/types/client.go]: Ah, cool, I thought I'd ask just in case :)

**Comment by thaJeztah** [comment]
LGTM

**Comment by vdemeester** [comment]
Feels LGTM 🐶 to me :wink:

File: daemon/daemon.go [hunk 2]
```diff
@@ -1013,8 +1013,8 @@ func (daemon *Daemon) changes(container *container.Container) ([]archive.Change,
 	return container.RWLayer.Changes()
 }
 
-// TagImage creates a tag in the repository reponame, pointing to the image named
-// imageName.
+// TagImage creates a tag in the repository newTag.Name(), pointing to the
```
Comment (aaronlehmann) [inline daemon/daemon.go]: I would say `newTag` instead of `newTag.Name()`

Comment (cr7pt0gr4ph7) [inline daemon/daemon.go]: Is this better?
```go
// TagImage creates a tag in the repository and with the name specified by newTag,
// pointing to the image named imageName (alternatively, imageName can also be an image ID).
func (daemon *Daemon) TagImage(newTag reference.Named, imageName string) error {
  // ...
}
```
I'm unsure about the right wording here. Suggestions welcome. :)

Comment (aaronlehmann) [inline daemon/daemon.go]: I think the "and" is extraneous. Otherwise, it definitely looks like an improvement.

**Comment by cr7pt0gr4ph7** [comment]
It seems that `ImageImportOptions.Changes`, which is mapped to the `changes` query parameter for `/images/create` is currently not implemented. i.e. the `changes` parameter is ignored by the server-side code.

An issue should probably be filed for this; I just wanted to note that this problem is out-of-scope for this PR.

**Comment by cr7pt0gr4ph7** [comment]
I incorporated the changes suggested by @aaronlehmann. Will squash into one commit after review.

**Comment by runcom** [comment]
This also needs a rebase @cr7pt0gr4ph7 :(

**Comment by cr7pt0gr4ph7** [comment]
I'll put the changes to everything under `api/types` into a separate pull request for [docker/engine-api](https://github.com/docker/engine-api), and squash everything else into one commit.

**Comment by cr7pt0gr4ph7** [comment]
Rebased onto current master.
The changes to `types.ImageCreateOptions` etc. have been separated into a docker/engine-api#23.

**Comment by thaJeztah** [comment]
LGTM

ping @runcom @aaronlehmann

File: daemon/daemon.go [hunk 3]
```diff
@@ -1013,8 +1013,8 @@ func (daemon *Daemon) changes(container *container.Container) ([]archive.Change,
 	return container.RWLayer.Changes()
 }
 
-// TagImage creates a tag in the repository reponame, pointing to the image named
-// imageName.
+// TagImage creates a tag in the repository and with the name specified by newTag,
```
Comment (aaronlehmann) [inline daemon/daemon.go]: I'm still finding this confusing. How about something like "TagImage creates the tag specified by newTag"?

**Comment by cpuguy83** [comment]
Carrying here: #19837 wit @aaronlehmann changes.

**Pull request closed without merging**